Numeric | 'Unit' isn't working inside a rule

This would fail:

```ruby
rule 'x' do
  every :second
  run { logger.info("X: #{0 | 'W'}") }
end
```

Or it could be a `changed` trigger or any other trigger apart from on_start

Error message:
```
21:57:43.268 [ERROR] [y.OpenHAB::DSL::Rules::AutomationRule] - wrong number of arguments (given 3, expected 1) (ArgumentError)
In rule: x
/openhab/conf/automation/jsr223/ruby/personal/test.rb:94:in `method_missing'
/openhab/conf/automation/jsr223/ruby/personal/test.rb:67:in `block in <main>'
```


This works, however:
```ruby
rule 'x' do
  on_start
  run { logger.info("X: #{0 | 'W'}") }
end
```

--- a/features/quantity_type.feature
+++ b/features/quantity_type.feature
@@ -17,6 +17,26 @@ Feature:  quantity_type
       | 50.0 \| '°F'     | 50.0 °F  |
       | 50.to_d \| '°F'  | 50.0 °F  |
 
+  Scenario Outline: QuantityType is constructuble with | from numeric from within rule
+    Given items:
+      | type   | name        | state |
+      | Number | NumberItem1 | 0     |
+    And code in a rules file
+      """
+        rule '| test' do
+          changed NumberItem1
+          run { logger.info("Result is #{(<quantity>).format('%.1f %unit%')}") }
+        end
+      """
+    When I deploy the rules file
+    And item "NumberItem1" state is changed to "1"
+    Then It should log 'Result is <result>' within 5 seconds
+    Examples:
+      | quantity       | result  |
+      | 50 \|'°F'      | 50.0 °F |
+      | 50.0 \|'°F'    | 50.0 °F |
+      | 50.to_d \|'°F' | 50.0 °F |
+
   Scenario Outline: QuantityType responds to math operations where operand is a quantity type
     Given code in a rules file
       """

--- a/lib/openhab/dsl/monkey_patch/ruby/number.rb
+++ b/lib/openhab/dsl/monkey_patch/ruby/number.rb
@@ -91,7 +91,7 @@ module OpenHAB
 
             return super unless other.is_a?(javax.measure.Unit)
 
-            QuantityType.new(to_d.to_java, other)
+            Types::QuantityType.new(to_d.to_java, other)
           end
         end
       end

